pyanaconda/ui/tui/spokes/installation_source.py

```diff
@@ -151,10 +151,20 @@ def completed(self):
         """Is the spoke complete?"""
         return self.ready and not self._error and self.payload.is_ready()
 
+    def setup(self, args=None):
+        """Set up the spoke right before it is used."""
+        super().setup(args)
+        # Show any error/warning messages from payload setup on entry
+        self._add_messages_widget_if_any()
```

Comment:
I wonder why you added the message also here in `setup`. Is there some case (some hang maybe?) when it shows the message? I'd think the `refresh` method being called every time on entering the spoke is enough. It seems `setup` is targeted used for synchronization and kind of spoke initialization for the session. There is a direct `print` in storage spoke but I wonder if adding a widget to the window in `setup` will actually make effect (be displayed to user).

Reply:
Hm I thought refresh is not triggered on frist enter, only on each follow up re-render.

If this is not the case, I will remove it from setup. I will try now again.

Reply:
@rvykydal you are right, I removed the add_messages_widget_if_any from the setup and left only in the refresh and there was not effect. The messages appear as desired when entering the spoke.